coordinator/src/main/scala/filodb.coordinator/ShardManager.scala

```diff
@@ -37,6 +37,14 @@ private[coordinator] final class ShardManager(settings: FilodbSettings,
   private val _coordinators = new mutable.LinkedHashMap[Address, ActorRef]
   private val _errorShardReassignedAt = new mutable.HashMap[DatasetRef, mutable.HashMap[Int, Long]]
 
+  private val _shardMetricAggregator = NamespaceCardinalityPublisher(
+    () => { _datasetInfo.map{ case (dsRef, _) => dsRef}.toIterator },
+    () => { _coordinators.head._2 }
+  )
+  if (settings.config.getBoolean("shard-key-level-ingestion-metrics-enabled")) {
+    _shardMetricAggregator.schedulePeriodicPublishJob()
+  }
```

Comment:
`NamespaceCardinalityPublisher` could be renamed to `TenantIngestionMetering`. We dont want concept of Namespace in class names.

Also, we should instantiate the publisher only if `shard-key-level-ingestion-metrics-enabled` is true. It could be done using an option.

Reply:
Done!